ci/ray_ci/test_upgrade_version.py

```diff
@@ -0,0 +1,198 @@
+from unittest import mock
+import sys
+import pytest
+import tempfile
+import os
+
+from ci.ray_ci.upgrade_version import (
+    list_java_files,
+    get_current_version,
+    upgrade_file_version,
+)
+
+
+def test_list_java_files():
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        os.mkdir(os.path.join(tmp_dir, "subdir_0"))
+        os.mkdir(os.path.join(tmp_dir, "subdir_1"))
+        os.mkdir(os.path.join(tmp_dir, "subdir_0/subdir_0_0"))
+        select_file_paths = [
+            "pom_template.xml",
+            "subdir_0/pom.xml",
+            "subdir_0/pom_template.xml",
+            "subdir_1/pom.xml",
+            "subdir_1/pom_template.xml",
+            "subdir_0/subdir_0_0/pom.xml",
+        ]
+        non_select_file_paths = [
+            "not_pom.xml",
+            "subdir_1/not_pom_template.xml",
+            "subdir_0/subdir_0_0/not_pom_template.xml",
+            "subdir_0/subdir_0_0/not_pom.xml",
+        ]
+        for file_path in select_file_paths + non_select_file_paths:
+            with open(os.path.join(tmp_dir, file_path), "w") as f:
+                f.write("")
+
+        assert list_java_files(tmp_dir) == sorted(
+            [os.path.join(tmp_dir, file_path) for file_path in select_file_paths]
+        )
+
+
+@mock.patch("ci.ray_ci.upgrade_version.get_check_output")
+def test_get_current_version(mock_check_output):
+    mock_check_output.side_effect = [
+        "3.0.0.dev0 a123456dc1d2egd345a6789f1e23d45b678c90ed",
+        "2.2.0 a123456dc1d2egd345a6789f1e23d45b678c90ed",
+    ]
+    # Test when version is default
+    assert get_current_version(tempfile.gettempdir()) == (
+        "3.0.0.dev0",
+        "2.0.0-SNAPSHOT",
+    )
+    # Test when version is different
+    assert get_current_version(tempfile.gettempdir()) == (
+        "2.2.0",
+        "2.2.0",
+    )
+
+
+def _prepare_file(file_path, java: bool = False):
+    """
+    Print to a file with version in the format of Java or non-Java files.
+    """
+    with open(file_path, "w") as f:
+        if java:
+            f.write("<version>2.0.0-SNAPSHOT</version>")
+        else:
+            f.write("version: 2.2.0.dev0")
+        f.flush()
+
+
+def _make_tmp_directories(tmp_dir):
+    directories = [
+        "ci",
+        "ci/ray_ci",
+        "python",
+        "python/ray",
+        "src",
+        "src/ray",
+        "src/ray/common",
+        "subdir_0",
+        "subdir_1",
+        "subdir_0/subdir_0_0",
+    ]
+    for dir in directories:
+        full_dir_path = os.path.join(tmp_dir, dir)
+        os.mkdir(full_dir_path)
+
+
+def test_upgrade_file_version():
+    main_version = "2.2.0.dev0"
+    java_version = "2.0.0-SNAPSHOT"
+    new_version = "1.1.1"
```

Comment:
change this to `2.3.2` or something that looks real?

Reply:
done!